lib/iris/coords.py

```diff
@@ -1478,32 +1619,74 @@ def _repr_other_metadata(self):
             result += ', circular=%r' % self.circular
         return result
 
-    @property
-    def points(self):
-        """The local points values as a read-only NumPy array."""
-        return self._points
-
-    @points.setter
-    def points(self, points):
-        points = np.array(points, ndmin=1)
-        # If points are already defined for this coordinate,
-        if hasattr(self, '_points') and self._points is not None:
-            # Check that setting these points wouldn't change self.shape
-            if points.shape != self.shape:
-                raise ValueError("New points shape must match existing points "
-                                 "shape.")
+    def _new_points_requirements(self, points):
+        """
+        Confirm that a new set of coord points adheres to the requirements for
+        :class:`~iris.coords.DimCoord` points, being:
+            * points are 1D,
+            * points are numeric, and
+            * points are monotonic.
 
-        # Checks for 1d, numeric, monotonic
+        """
         if points.ndim != 1:
             raise ValueError('The points array must be 1-dimensional.')
         if not np.issubdtype(points.dtype, np.number):
             raise ValueError('The points array must be numeric.')
         if len(points) > 1 and not iris.util.monotonic(points, strict=True):
             raise ValueError('The points array must be strictly monotonic.')
+
+    @property
+    def points(self):
+        """The local points values as a read-only NumPy array."""
+        return self._points_dm.data
+
+    @points.setter
+    def points(self, points):
+        # Checks for 1d, numeric, monotonic
+        self._new_points_requirements(points)
+
+        # We must realise points to check monotonicity.
+        points = as_concrete_data(points, result_dtype=self.dtype)
+
         # Make the array read-only.
         points.flags.writeable = False
 
-        self._points = points
+        self._points_dm.data = points
```

Comment:
There's a lot of processing in the `DataManager.data` setter and I'm not sure if it guarantees that the array assigned will be the same object as the one passed in. It might be better to set `self._points_dm.data.writeable = False` after the assignment, rather than on the `points` array before.